syntax = "proto2";

package addressbook;
service RPCService {

  rpc GetPersonByPhoneNumber(PhoneNumber) returns (Person) {}
  rpc EditPeople(stream Person) returns (ResponseEdit) {}
  rpc ListPeopleByPhoneType(PhoneNumber) returns (stream Person) {}
  rpc GetPeopleById(stream RequestId) returns (stream Person) {}
}
enum PhoneType {
    MOBILE = 0;
    HOME = 1;
    WORK = 2;
}
message PhoneNumber {
    optional string number = 1;
    optional PhoneType type = 2 [default = HOME];
}

message Person {
  optional string name = 1;
  required int32 id = 2;
  optional string email = 3;

  repeated PhoneNumber phones = 4;
}

message AddressBook {
  repeated Person people = 1;
}
message ResponseEdit {
  required bool result = 1;
}
message RequestId {
  required int32 id = 1;
}